Meta/CMake/common_compile_options.cmake

```diff
@@ -5,42 +5,54 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 set(CMAKE_COLOR_DIAGNOSTICS ON)
 
-add_compile_options(-Wall)
-add_compile_options(-Wextra)
+# FIXME: when compiing using cl-clang, the code tries to compile c++98 mode, this generates too much warnings
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang$" AND NOT CMAKE_CXX_SIMULATE_ID MATCHES "MSVC")
+    add_compile_options(-Wall)
+    add_compile_options(-Wextra)
+endif()
 
 add_compile_options(-Wno-invalid-offsetof)
 
 add_compile_options(-Wno-unknown-warning-option)
 add_compile_options(-Wno-unused-command-line-argument)
 
-add_compile_options(-fno-exceptions)
-
-add_compile_options(-ffp-contract=off)
-
-if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "18")
-    add_compile_options(-Wpadded-bitfield)
-endif()
 
 if (NOT CMAKE_HOST_SYSTEM_NAME MATCHES SerenityOS)
     # FIXME: Something makes this go crazy and flag unused variables that aren't flagged as such when building with the toolchain.
     #        Disable -Werror for now.
-    add_compile_options(-Werror)
+    # add_compile_options(-Werror)
 endif()
 
-if (CMAKE_CXX_COMPILER_ID MATCHES "Clang$")
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "18")
+    add_compile_options(-Wpadded-bitfield)
+endif()
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND NOT CMAKE_CXX_SIMULATE_ID  MATCHES "MSVC")
     # Clang's default constexpr-steps limit is 1048576(2^20), GCC doesn't have one
     add_compile_options(-fconstexpr-steps=16777216)
+    add_compile_options(-fno-exceptions)
 
     add_compile_options(-Wno-implicit-const-int-float-conversion)
     add_compile_options(-Wno-user-defined-literals)
     add_compile_options(-Wno-vla-cxx-extension)
+    add_compile_options(-ffp-contract=off)
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    add_compile_options(-ffp-contract=off)
+
     # Only ignore expansion-to-defined for g++, clang's implementation doesn't complain about function-like macros
     add_compile_options(-Wno-expansion-to-defined)
     add_compile_options(-Wno-literal-suffix)
+    add_compile_options(-fno-exceptions)
 
     # FIXME: This warning seems useful but has too many false positives with GCC 13.
     add_compile_options(-Wno-dangling-reference)
+elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang$" AND CMAKE_CXX_SIMULATE_ID MATCHES "MSVC")
```

Comment:
we must have no exceptions. EH- at least.

Reply:
Refactored the code, look a line 11. Now on linux I have the same compilation flags as before.